kernel/Kconfig

```diff
@@ -330,6 +330,10 @@ config SYSTEM_WORKQUEUE_PRIORITY
 	default -1
 	default  0 if !COOP_ENABLED
 	default -2 if COOP_ENABLED && !PREEMPT_ENABLED
+	help
+	  By default, system work queue priority is the lowest cooperative
+	  priority. This means that any work handler, once started, won't
+	  be preempted by any other thread until finished.
```

Comment:
"until finished or it blocks" might be more accurate.

Reply:
True, though there's an implicit promise (probably worth documenting, heh) that a system work_q item isn't going to sleep or block, otherwise we have different problems.

Reply:
> there's an implicit promise (probably worth documenting, heh) that a system work_q item isn't going to sleep or block

That's what I had in mind, because otherwise there would be pretty big list of "if" and "unless", if describing all the possibilities. I can add that "or blocks" if desired of course.